### PR TITLE
UGENE-7118. Fix tree_nodes_creation_test_0001 GUI test

### PR DESCRIPTION
**Pull request by yalgaer**

### CODE REVIEW

File: src/plugins/GUITestBase/src/GTUtilsWorkflowDesigner.h [hunk 1]
```diff
@@ -144,16 +141,35 @@ class GTUtilsWorkflowDesigner {
...
-    static void setDatasetInputFolder(HI::GUITestOpStatus &os, QString filePath);
-    static void setDatasetInputFolders(HI::GUITestOpStatus &os, const QStringList &dirPaths);
+    /**
+     * Sets input folder with path "filePath" to dataset.
+     * If datasetWidget is not provided it is looked up the current WD window.
+     */
+    static void setDatasetInputFolder(HI::GUITestOpStatus &os, const QString &filePath, QWidget *datasetWidget = nullptr);
+
+    /**
+     * Adds all folders form the to the active dataset.
```
Comment (evelinabis) [inline src/plugins/GUITestBase/src/GTUtilsWorkflowDesigner.h]: Typo?

File: src/plugins/GUITestBase/src/GTUtilsWorkflowDesigner.h [hunk 2]
```diff
@@ -101,25 +98,25 @@ class GTUtilsWorkflowDesigner {
...
-    static void addSample(HI::GUITestOpStatus &os, const QString &sampName, QWidget const *const parentWidget = NULL);
-    static void addAlgorithm(HI::GUITestOpStatus &os, QString algName, bool exactMatch = false, bool useDragAndDrop = false);
+    static void addSample(HI::GUITestOpStatus &os, const QString &sampName, QWidget *parentWidget = nullptr);
+    static void addAlgorithm(HI::GUITestOpStatus &os, const QString &algName, bool exactMatch = false, bool useDragAndDrop = false);
     static WorkflowProcessItem *addElement(HI::GUITestOpStatus &os, const QString &algName, bool exactMatch = false);
     static WorkflowProcessItem *addElementByUsingNameFilter(HI::GUITestOpStatus &os, const QString &elementName, bool exactMatch = false);
 
     //returns center of worker on workflow scene in global coordinates
-    static QPoint getItemCenter(HI::GUITestOpStatus &os, QString itemName);
+    static QPoint getItemCenter(HI::GUITestOpStatus &os, const QString& itemName);
```
Comment (evelinabis) [inline src/plugins/GUITestBase/src/GTUtilsWorkflowDesigner.h]: Sometimes you write `const QString&`, sometimes `const QString &`

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp [hunk 3]
```diff
@@ -3064,14 +3060,16 @@ void GUITestBasePlugin::registerTests(UGUITestBase *guiTestBase) {
...
+    REGISTER_TEST_L(GUITest_common_scenarios_workflow_dashboard::tree_nodes_creation_test_0001, labels({Metagenomics, Linux})); // Needs full taxonomy data (Not available in Nightly).
     REGISTER_TEST(GUITest_common_scenarios_workflow_dashboard::tree_nodes_creation_test_0002);
     REGISTER_TEST(GUITest_common_scenarios_workflow_dashboard::tree_nodes_creation_test_0003);
     REGISTER_TEST(GUITest_common_scenarios_workflow_dashboard::tree_nodes_creation_test_0004);
 
     REGISTER_TEST(GUITest_common_scenarios_workflow_dashboard::tool_launch_nodes_test_0001);
     REGISTER_TEST(GUITest_common_scenarios_workflow_dashboard::tool_launch_nodes_test_0002);
     REGISTER_TEST(GUITest_common_scenarios_workflow_dashboard::tool_launch_nodes_test_0003);
-
+    REGISTER_TEST_L(GUITest_common_scenarios_workflow_dashboard::tool_launch_nodes_test_0014, labels({Metagenomics, Linux})); // Needs full taxonomy data (Not available in Nightly).
```
Comment (evelinabis) [inline src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: Out of order

**Review by evelinabis — DISMISSED** [review]

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp [hunk 4]
```diff
@@ -3064,14 +3060,16 @@ void GUITestBasePlugin::registerTests(UGUITestBase *guiTestBase) {
     REGISTER_TEST(GUITest_common_scenarios_workflow_dashboard::misc_test_0002);
     REGISTER_TEST(GUITest_common_scenarios_workflow_dashboard::misc_test_0003);
     REGISTER_TEST(GUITest_common_scenarios_workflow_dashboard::misc_test_0004);
+
+    REGISTER_TEST_L(GUITest_common_scenarios_workflow_dashboard::tree_nodes_creation_test_0001, labels({Metagenomics, Linux})); // Needs full taxonomy data (Not available in Nightly).
```
Comment (DmitriiSukhomlinov) [inline src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: As far as I know, the "Metagenomics" suite doesn't use this labels right now, test should be added to "test/general/metagenomic_gui_tests_list.txt"

**Review by DmitriiSukhomlinov — DISMISSED** [review]
The code is OK, but tests should pass before the merge. Please, provide a link to the passed test run after they will finish.

File: src/plugins/GUITestBase/src/GTUtilsWorkflowDesigner.h (hunk 1 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/GTUtilsWorkflowDesigner.h]: Fixed

File: src/plugins/GUITestBase/src/GTUtilsWorkflowDesigner.h (hunk 2 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/GTUtilsWorkflowDesigner.h]: I always write String& but the UGENE code style requires String &. Sometimes I forgot to run a formatter. Fixed

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp (hunk 4 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: Ok, I will add tests to that list

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp (hunk 3 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: Fixed

**Comment by yalgaer** [comment]
> The code is OK, but tests should pass before the merge. Please, provide a link to the passed test run after they will finish.

The runs are in the Teamcity. Check for the UGENE-7118 branch

**Pull request merged**